compiler-core/src/parse.rs

```diff
@@ -2176,6 +2195,19 @@ where
         } else {
             (vec![], end)
         };
+
+        // check if all constructors are deprecated if so error
+        if constructors.len() > 0 // prevent checking an empty type `type wobble` which will always be true
+            && constructors
+                .iter()
+                .all(|record| record.deprecation.is_deprecated())
+        {
+            return parse_error(
+                ParseErrorType::AllVariantRecordConstructorDeprecated,
+                SrcSpan { start, end },
+            );
```

Comment:
I think this error should happen during the analysis phase and not in the parser, if an error happens during parsing it would stop the compiler from being able to analyse the rest of the code

Reply:
Yup!